### PR TITLE
Use assertj fluent style in flowable-engine module.

### PR DESCRIPTION
**Pull request by dbmalkovsky**
Various test files in the `org.flowable.standalone` directory.

Specifically test files in the `initialization`, `interceptor`, `jpa`, `parsing`, `rules`, `scripting`, and 
`validation` packages.

### CODE REVIEW

File: modules/flowable-engine/src/test/java/org/flowable/standalone/initialization/NoDbConnectionTest.java [hunk 1]
```diff
@@ -25,21 +27,8 @@
...
-        if (e == null) {
-            return false;
-        }
-        if (e instanceof SQLException) {
-            return true;
-        }
-        return containsSqlException(e.getCause());
+        assertThatThrownBy(() -> ProcessEngineConfiguration
+                .createProcessEngineConfigurationFromResource("org/flowable/standalone/initialization/nodbconnection.flowable.cfg.xml").buildProcessEngine())
+                .hasCauseInstanceOf(SQLException.class);
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/standalone/initialization/NoDbConnectionTest.java:32]: This assertion is not the same as before. Before it was checking if the exception itself is `SQLException` or there is a `SQLException` in the stacktrace. I don't know why that was done like that, might be needed for the other DBs perhaps.

File: modules/flowable-engine/src/test/java/org/flowable/standalone/initialization/ProcessEngineInitializationTest.java [hunk 2]
```diff
@@ -73,18 +74,15 @@ public void testVersionMismatch() {
...
-            assertTextPresent("version mismatch", e.getMessage());
-            assertEquals("25.7", e.getDbVersion());
-            assertEquals(ProcessEngine.VERSION, e.getLibraryVersion());
-        }
+        // now we can see what happens if when a process engine is being
+        // build with a version mismatch between library and db tables
+        assertThatThrownBy(() -> ProcessEngineConfiguration
+                .createProcessEngineConfigurationFromResource("org/flowable/standalone/initialization/notables.flowable.cfg.xml")
+                .setDatabaseSchemaUpdate(ProcessEngineConfiguration.DB_SCHEMA_UPDATE_FALSE).buildProcessEngine())
+                .isExactlyInstanceOf(FlowableWrongDbException.class)
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/standalone/initialization/ProcessEngineInitializationTest.java:82]: Use `isInstanceOfSatisfying` so we can test the `FlowableWrongDbException#getDbVersion` and `FlowableWrongDbException#getLibraryVersion`, not just that the message has it.

File: modules/flowable-engine/src/test/java/org/flowable/standalone/jpa/HistoricJPAVariableTest.java [hunk 3]
```diff
@@ -91,8 +93,8 @@ public void testGetJPAEntityAsHistoricVariable() {
                 .processInstanceId(processInstanceId).variableName("simpleEntityFieldAccess").singleResult();
 
         Object value = historicVariableInstance.getValue();
-        assertTrue(value instanceof FieldAccessJPAEntity);
-        assertEquals(((FieldAccessJPAEntity) value).getValue(), simpleEntityFieldAccess.getValue());
+        assertThat(value instanceof FieldAccessJPAEntity).isTrue();
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/standalone/jpa/HistoricJPAVariableTest.java:96]: use `assertThat(value).isInstanceOf(FieldAccessJPAEntity.class);`

File: modules/flowable-engine/src/test/java/org/flowable/standalone/rules/BusinessRuleTaskTest.java [hunk 4]
```diff
@@ -26,17 +28,18 @@
     @Deployment
     public void testBusinessRuleTask() {
         ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("customRule");
-        assertEquals("test2", runtimeService.getVariable(processInstance.getId(), "test"));
+        assertThat(runtimeService.getVariable(processInstance.getId(), "test")).isEqualTo("test2");
 
-        assertEquals(1, CustomBusinessRuleTask.ruleInputVariables.size());
-        assertEquals("order", CustomBusinessRuleTask.ruleInputVariables.get(0).getExpressionText());
+        assertThat(CustomBusinessRuleTask.ruleInputVariables);
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/standalone/rules/BusinessRuleTaskTest.java:33]: This is a dangling assertion

File: modules/flowable-engine/src/test/java/org/flowable/standalone/rules/RulesDeployerTest.java [hunk 5]
```diff
@@ -41,16 +43,16 @@ public void testRulesDeployment() {
         variableMap.put("order", order);
 
         ProcessInstance processInstance = runtimeService.startProcessInstanceByKey("rulesDeployment", variableMap);
-        assertNotNull(processInstance);
-        assertTrue(processInstance.getProcessDefinitionId().startsWith("rulesDeployment:1"));
+        assertThat(processInstance).isNotNull();
+        assertThat(processInstance.getProcessDefinitionId().startsWith("rulesDeployment:1"));
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/standalone/rules/RulesDeployerTest.java:47]: Dangling assertion

File: modules/flowable-engine/src/test/java/org/flowable/standalone/validation/DefaultProcessValidatorTest.java [hunk 6]
```diff
@@ -340,40 +336,41 @@ void testEventSubProcessWithBoundary() {
...
 
-        Assert.assertNotNull(error.getActivityId());
-        Assert.assertNotNull(error.getActivityName());
+        assertThat(error.getActivityId()).isNotNull();
+        assertThat(error.getActivityName()).isNotNull();
 
-        Assert.assertTrue(error.getActivityId().length() > 0);
-        Assert.assertTrue(error.getActivityName().length() > 0);
+        assertThat(error.getActivityId().length() > 0).isTrue();
+        assertThat(error.getActivityName().length() > 0).isTrue();
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/standalone/validation/DefaultProcessValidatorTest.java:351]: Can be replaces with 

```
assertThat(error.getActivityId()).isNotEmpty();
assertThat(error.getActivityName()).isNotEmpty();
```

File: modules/flowable-engine/src/test/java/org/flowable/standalone/validation/DefaultProcessValidatorTest.java [hunk 7]
```diff
@@ -340,40 +336,41 @@ void testEventSubProcessWithBoundary() {
...
-        Assert.assertNotNull(error.getValidatorSetName());
-        Assert.assertNotNull(error.getProblem());
-        Assert.assertNotNull(error.getDefaultDescription());
-        Assert.assertTrue(error.getXmlLineNumber() > 0);
-        Assert.assertTrue(error.getXmlColumnNumber() > 0);
+        assertThat(error.getValidatorSetName()).isNotNull();
+        assertThat(error.getProblem()).isNotNull();
+        assertThat(error.getDefaultDescription()).isNotNull();
+        assertThat(error.getXmlLineNumber() > 0).isTrue();
+        assertThat(error.getXmlColumnNumber() > 0).isTrue();
```
Comment (filiphr) [inline modules/flowable-engine/src/test/java/org/flowable/standalone/validation/DefaultProcessValidatorTest.java:359]: Can be replaces with 

```
assertThat(error.getXmlLineNumber()).isPositive();
assertThat(error.getXmlColumnNumber()).isPositive();
```

File: modules/flowable-engine/src/test/java/org/flowable/standalone/initialization/NoDbConnectionTest.java (hunk 1 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/standalone/initialization/NoDbConnectionTest.java:32]: I think this does the same thing as the manually coded routine; it looks for some exception in the traceback that satisfies a `SQLException` directly or indirectly as an extension.

File: modules/flowable-engine/src/test/java/org/flowable/standalone/initialization/ProcessEngineInitializationTest.java (hunk 2 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/standalone/initialization/ProcessEngineInitializationTest.java:82]: Done; thanks.

File: modules/flowable-engine/src/test/java/org/flowable/standalone/jpa/HistoricJPAVariableTest.java (hunk 3 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/standalone/jpa/HistoricJPAVariableTest.java:96]: done; one that I missed :(

File: modules/flowable-engine/src/test/java/org/flowable/standalone/rules/BusinessRuleTaskTest.java (hunk 4 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/standalone/rules/BusinessRuleTaskTest.java:33]: Bad copy-n-paste; fixed.

File: modules/flowable-engine/src/test/java/org/flowable/standalone/rules/RulesDeployerTest.java (hunk 5 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/standalone/rules/RulesDeployerTest.java:47]: Fixed; thanks

File: modules/flowable-engine/src/test/java/org/flowable/standalone/validation/DefaultProcessValidatorTest.java (hunk 6 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/standalone/validation/DefaultProcessValidatorTest.java:351]: Fixed; thanks; guess I was getting a little cross eyed by this time.

File: modules/flowable-engine/src/test/java/org/flowable/standalone/validation/DefaultProcessValidatorTest.java (hunk 7 above)
Comment (dbmalkovsky) [reply modules/flowable-engine/src/test/java/org/flowable/standalone/validation/DefaultProcessValidatorTest.java:359]: Ditto.

**Comment by dbmalkovsky** [comment]
@filiphr I use `isExactlyInstanceOf` so it is obvious if the exception type changes at some point.

I think I've responded to all your comments.  Thanks for the detailed review.

**Pull request merged**